api/converter_utils.go

```diff
@@ -624,6 +615,15 @@ func (si *ServerImplementation) transactionParamsToTransactionFilter(params gene
 	// Boolean
 	filter.RekeyTo = params.RekeyTo
 
+	// filter Algos or Asset but not both.
+	if filter.AssetID != 0 || filter.TypeEnum == idb.TypeEnumAssetTransfer {
+		filter.AssetAmountLT = params.CurrencyLessThan
+		filter.AssetAmountGT = params.CurrencyGreaterThan
+	} else {
+		filter.AlgosLT = params.CurrencyLessThan
+		filter.AlgosGT = params.CurrencyGreaterThan
+	}
+
```

Comment:
this block is just being moved and it doesn't change the method's behavior, right?

Reply:
I changed the first condition, but needed to move it down so that `filter.TypeEnum` is initialized.